diff --git a/build/m4/shamrock/mono.m4 b/build/m4/shamrock/mono.m4
index c40ecbf..b0699fb 100755
--- a/build/m4/shamrock/mono.m4
+++ b/build/m4/shamrock/mono.m4
@@ -57,7 +57,8 @@ AC_DEFUN([_SHAMROCK_CHECK_MONO_GAC_ASSEMBLIES],
 		AC_MSG_CHECKING([for Mono $2 GAC for $asm.dll])
 		if test \
 			-e "$($PKG_CONFIG --variable=libdir $1)/mono/$2/$asm.dll" -o \
-			-e "$($PKG_CONFIG --variable=prefix $1)/lib/mono/$2/$asm.dll"; \
+			-e "$($PKG_CONFIG --variable=prefix $1)/lib/mono/$2/$asm.dll" -o \
+			-e "/usr/lib/mono/xbuild-frameworks/.NETPortable/v$2/$asm.dll"; \
 			then \
 			AC_MSG_RESULT([found])
 		else
@@ -91,4 +92,3 @@ AC_DEFUN([SHAMROCK_CHECK_MONO2_4_0_GAC_ASSEMBLIES],
 [
 	_SHAMROCK_CHECK_MONO_GAC_ASSEMBLIES(mono-2, 4.0, $*)
 ])
-
