--- src/Frontend-GNOME/FindGroupChatDialog.cs	2008-12-26 16:42:42.000000000 -0700
+++ src/Frontend-GNOME/FindGroupChatDialog.cs	2009-01-12 16:55:35.000000000 -0700
@@ -115,12 +115,14 @@
                 f_ListStore.Clear();
                 CancelFindThread();
                 
+                EventHandler d = delegate {
+                    GdkWindow.Cursor = new Gdk.Cursor(Gdk.CursorType.Watch);
+                };
+
                 GroupChatModel filter =  new GroupChatModel(null, nameFilter, null);
                 f_FindThread = new Thread(new ThreadStart(delegate {
                     try {
-                        Gtk.Application.Invoke(delegate {
-                            GdkWindow.Cursor = new Gdk.Cursor(Gdk.CursorType.Watch);
-                        });
+                        Gtk.Application.Invoke(d);
                         
                         IList<GroupChatModel> chats = f_ProtocolManager.FindGroupChats(filter);
                         
