--- b/src/NLog/NLog.mono.csproj
+++ a/src/NLog/NLog.mono.csproj
@@ -435,6 +435,17 @@
     <CodeAnalysisDictionary Include="CustomDictionary.xml" />
   </ItemGroup>
   <Import Project="$(MSBuildToolsPath)\Microsoft.CSharp.targets" />
+  <Import Project="/usr/lib/mono/4.5/MSBuild.Community.Tasks.Targets" />
+  <Target Name="BeforeBuild">
+    <PropertyGroup Condition=" '$(VersionNumber)' == '' ">
+      <VersionNumber>1.0.0.0</VersionNumber>
+    </PropertyGroup>
+    <AssemblyInfo CodeLanguage="C#"
+      OutputFile="Properties/AssemblyInfo.cs" 
+      AssemblyVersion="$(VersionNumber)" 
+      AssemblyFileVersion="$(VersionNumber)"
+    />
+  </Target>
   <ItemGroup>
     <None Include="Logger.tt">
       <Generator>TextTemplatingFileGenerator</Generator>
diff -r -u a/src/NLog.Extended/NLog.Extended.mono.csproj b/src/NLog.Extended/NLog.Extended.mono.csproj
--- a/src/NLog.Extended/NLog.Extended.mono.csproj	2016-11-09 01:54:55.000000000 +0300
+++ b/src/NLog.Extended/NLog.Extended.mono.csproj	2016-11-13 15:54:01.909919749 +0300
@@ -71,6 +71,7 @@
     <CodeAnalysisDictionary Include="CustomDictionary.xml" />
   </ItemGroup>
   <Import Project="$(MSBuildToolsPath)\Microsoft.CSharp.targets" />
+  <Import Project="/usr/lib/mono/4.5/MSBuild.Community.Tasks.Targets" />
   <ItemGroup>
     <ProjectReference Include="..\NLog\NLog.mono.csproj">
       <Project>{E7AB20BF-6920-442A-B876-CC05BC5CEC79}</Project>
@@ -81,5 +82,13 @@
     <ItemGroup Condition="Exists('$(BuildInfoFile)')">
       <Compile Include="$(BuildInfoFile)" />
     </ItemGroup>
+    <PropertyGroup Condition=" '$(VersionNumber)' == '' ">
+        <VersionNumber>1.0.0.0</VersionNumber>
+    </PropertyGroup>
+    <AssemblyInfo CodeLanguage="C#"
+      OutputFile="Properties/AssemblyInfo.cs" 
+      AssemblyVersion="$(VersionNumber)" 
+      AssemblyFileVersion="$(VersionNumber)"
+      />
   </Target>
 </Project>
