diff -NrU5 NLog-1.0.orig/NLog.build NLog-1.0/NLog.build
--- NLog-1.0.orig/NLog.build	2009-02-04 22:22:45.000000000 +0100
+++ NLog-1.0/NLog.build	2009-02-04 22:25:03.000000000 +0100
@@ -148,11 +148,11 @@
 
         <property name="nlog.version" value="${nlog.version.major}.${nlog.version.minor}.${nlog.version.revision}.${nlog.version.build}" />
     </target>
 
     <target name="NLog" depends="configure, update-build-number">
-        <csc target="library" output="${nlog.dir}/NLog.dll" doc="${nlog.dir}/NLog.xml" define="${nlog.define}" debug="${nlog.debug}" optimize="${nlog.optimize}" warninglevel="${csc.warninglevel}" warnaserror="true">
+        <csc target="library" output="${nlog.dir}/NLog.dll" doc="${nlog.dir}/NLog.xml" define="${nlog.define}" debug="${nlog.debug}" optimize="${nlog.optimize}" warninglevel="${csc.warninglevel}" warnaserror="false">
             <sources basedir="src">
                 <include name="NLog/**/*.cs" />
             </sources>
             <references>
                 <include name="mscorlib.dll" />
@@ -719,11 +719,11 @@
                 <namespaces>
                     <namespace prefix="ms" uri="urn:schemas-microsoft-com:asm.v1" />
                 </namespaces>
             </xmlpoke>
 
-            <exec program="${path::combine(framework::get-framework-directory(framework::get-target-framework()),'al')}"
+            <exec program="/usr/bin/al"
                 commandline="/nologo /link:Policy.${nlog.version.major}.${nlog.version.minor}.NLog.xml /out:Policy.${nlog.version.major}.${nlog.version.minor}.NLog.dll /keyfile:${path::combine(project::get-base-directory(),'src/NLog.snk')} /version:1.0.0.0"
                 workingdir="${nlog.dir}" />
         </if>
     </target>
 
