--- configure.in	2008-02-06 19:12:17.000000000 +0100
+++ configure.in.new	2008-03-20 14:37:13.000000000 +0100
@@ -43,26 +43,39 @@
 AC_SUBST(GTK_SHARP_LIBS)
 
 PKG_CHECK_MODULES(GTKHTML_SHARP, gtkhtml-sharp-3.14, enable_gtkhtml=yes, enable_gtkhtml=no)
-if test "x$enable_gtkhtml" = "xyes"; then
-CSHARP_FLAGS="-d:GTKHTML_SHARP_3_14"
-else
-PKG_CHECK_MODULES(GTKHTML_SHARP, gtkhtml-sharp-2.0, enable_gtkhtml=yes, enable_gtkhtml=no)
-CSHARP_FLAGS=
+dnl PKG_CHECK_MODULES(GTKHTML_SHARP, gtkhtml-sharp-3.14, enable_gtkhtml=yes, enable_gtkhtml=no)
+dnl if test "x$enable_gtkhtml" = "xyes"; then
+dnl CSHARP_FLAGS="-d:GTKHTML_SHARP_3_14"
+dnl else
+AC_ARG_ENABLE(gtkhtml,
+	      AC_HELP_STRING([--enable-gtkhtml], [Use GTKHTML Browser]),,
+	      enable_gtkhtml=yes)
+
+if test "x$enable_gtkhtml" = "xyes" ; then
+	PKG_CHECK_MODULES(GTKHTML_SHARP, gtkhtml-sharp-2.0)
+
+	AC_SUBST(GTKHTML_SHARP_LIBS)
+	AC_SUBST(CSHARP_FLAGS)
 fi
-AC_SUBST(GTKHTML_SHARP_LIBS)
-AC_SUBST(CSHARP_FLAGS)
-AM_CONDITIONAL(ENABLE_GTKHTML, test "x$enable_gtkhtml" = "xyes")
+dnl fi
 
-PKG_CHECK_MODULES(GECKO_SHARP, gecko-sharp-2.0, enable_gecko=yes, enable_gecko=no)
-AC_SUBST(GECKO_SHARP_LIBS)
-AM_CONDITIONAL(ENABLE_GECKO, test "x$enable_gecko" = "xyes")
+AM_CONDITIONAL(ENABLE_GTKHTML, test "x$enable_gtkhtml" = "xyes")
 
-if test "x$enable_gecko" = "xyes"; then
-   # get the mozilla home directory
-   MOZILLA_HOME="`$PKG_CONFIG --variable=libdir mozilla-gtkmozembed`"
-   AC_SUBST(MOZILLA_HOME)
+AC_ARG_ENABLE(mozilla,
+	      AC_HELP_STRING([--enable-mozilla], [Use Gecko Browser]),,
+	      enable_gecko=yes)
+
+if test "x$enable_gecko"= "xyes" ; then
+	PKG_CHECK_MODULES(GECKO_SHARP, gecko-sharp-2.0)
+
+	AC_SUBST(GECKO_SHARP_LIBS)
+
+	# get the mozilla home directory
+	MOZILLA_HOME="`$PKG_CONFIG --variable=libdir mozilla-gtkmozembed`"
+	AC_SUBST(MOZILLA_HOME)
 fi
 
+AM_CONDITIONAL(ENABLE_GECKO, test "x$enable_gecko" = "xyes")
 
 if test "x$enable_gtkhtml$enable_gecko" = "xnono" ; then
    AC_MSG_ERROR([no HTML display found. You need either gtkhtml or gecko])

--- docbrowser/monodoc.in	2008-03-10 23:34:01.000000000 +0100
+++ docbrowser/monodoc.in.new	2008-03-20 14:27:26.000000000 +0100
@@ -85,8 +85,10 @@
     MOZILLA_HOME=$MOZILLA_FIVE_HOME
 elif [ $(which mozilla 2> /dev/null) ] && grep MOZILLA_FIVE_HOME= "$(which mozilla)" > /dev/null ; then
 	MOZILLA_HOME=$(grep MOZILLA_FIVE_HOME= $(which mozilla) | cut -d '"' -f 2)
-elif [ $(which firefox 2> /dev/null) ] && grep MOZILLA_FIVE_HOME= "$(which firefox)" > /dev/null ; then
-	MOZILLA_HOME=$(grep MOZILLA_FIVE_HOME= $(which firefox) | cut -d '"' -f 2)
+elif [ $(which firefox 2> /dev/null) ] && grep MOZILLA_LIBDIR= "$(which firefox)" > /dev/null ; then
+	MOZILLA_HOME=$(grep MOZILLA_LIBDIR= $(which firefox) | cut -d '=' -f 2)
+elif [ $(which seamonkey 2> /dev/null) ] && grep MOZILLA_LIBDIR= "$(which seamonkey)" > /dev/null ; then
+	MOZILLA_HOME=$(grep MOZILLA_LIBDIR= $(which seamonkey) | cut -d '=' -f 2)
 fi
 
 if [ -n $LD_LIBRARY_PATH ]; then
