Only in mono-tools-1.1.17: autom4te.cache
diff -aur --exclude=Makefile.in --exclude=aclocal.m4 --exclude=configure mono-tools-1.1.17-orig/configure.in mono-tools-1.1.17/configure.in
--- mono-tools-1.1.17-orig/configure.in	2006-10-04 00:13:30.000000000 -0700
+++ mono-tools-1.1.17/configure.in	2006-10-28 05:28:17.000000000 -0700
@@ -42,21 +42,32 @@
 AC_SUBST(GTK_SHARP_LIBS)
 
 
-PKG_CHECK_MODULES(GTKHTML_SHARP, gtkhtml-sharp-2.0, enable_gtkhtml=yes, enable_gtkhtml=no)
-AC_SUBST(GTKHTML_SHARP_LIBS)
+AC_ARG_ENABLE(gtkhtml,
+              AC_HELP_STRING([--enable-gtkhtml], [Use GTKHTML Browser]),
+              enable_gtkhtml="$enableval", enable_gtkhtml=yes)
+
+if test "x$enable_gtkhtml" = "xyes" ; then
+   PKG_CHECK_MODULES(GTKHTML_SHARP, gtkhtml-sharp-2.0)
+   AC_SUBST(GTKHTML_SHARP_LIBS)
+fi
+
 AM_CONDITIONAL(ENABLE_GTKHTML, test "x$enable_gtkhtml" = "xyes")
 
 
-PKG_CHECK_MODULES(GECKO_SHARP, gecko-sharp-2.0, enable_gecko=yes, enable_gecko=no)
-AC_SUBST(GECKO_SHARP_LIBS)
-AM_CONDITIONAL(ENABLE_GECKO, test "x$enable_gecko" = "xyes")
+AC_ARG_ENABLE(mozilla,
+              AC_HELP_STRING([--enable-mozilla], [Use Gecko Browser]),
+              enable_gecko="$enableval", enable_gecko=yes)
+
+if test "x$enable_gecko" = "xyes" ; then
+   PKG_CHECK_MODULES(GECKO_SHARP, gecko-sharp-2.0)
+   AC_SUBST(GECKO_SHARP_LIBS)
 
-if test "x$enable_gecko" = "xyes"; then
-   # get the mozilla home directory
    MOZILLA_HOME="`$PKG_CONFIG --variable=libdir mozilla-gtkmozembed`"
    AC_SUBST(MOZILLA_HOME)
 fi
 
+AM_CONDITIONAL(ENABLE_GECKO, test "x$enable_gecko" = "xyes")
+
 
 if test "x$enable_gtkhtml$enable_gecko" = "xnono" ; then
    AC_MSG_ERROR([no HTML display found. You need either gtkhtml or gecko])
Only in mono-tools-1.1.17-orig/docbrowser: AssemblyInfo.cs
diff -aur --exclude=Makefile.in --exclude=aclocal.m4 --exclude=configure mono-tools-1.1.17-orig/docbrowser/Makefile.am mono-tools-1.1.17/docbrowser/Makefile.am
--- mono-tools-1.1.17-orig/docbrowser/Makefile.am	2006-10-04 00:13:30.000000000 -0700
+++ mono-tools-1.1.17/docbrowser/Makefile.am	2006-10-28 04:32:22.000000000 -0700
@@ -1,5 +1,7 @@
 monodocdir=$(prefix)/lib/monodoc
+if ENABLE_GTKHTML
 noinst_DATA = admin.exe
+endif
 
 
 if ENABLE_GECKO
diff -aur --exclude=Makefile.in --exclude=aclocal.m4 --exclude=configure mono-tools-1.1.17-orig/docbrowser/monodoc.in mono-tools-1.1.17/docbrowser/monodoc.in
--- mono-tools-1.1.17-orig/docbrowser/monodoc.in	2006-10-04 00:13:30.000000000 -0700
+++ mono-tools-1.1.17/docbrowser/monodoc.in	2006-10-28 07:50:20.000000000 -0700
@@ -69,8 +69,10 @@
         MOZILLA_HOME=@MOZILLA_HOME@
 elif [ $(which mozilla 2> /dev/null) ] && grep MOZILLA_FIVE_HOME= "$(which mozilla)" > /dev/null ; then
 	MOZILLA_HOME=$(grep MOZILLA_FIVE_HOME= $(which mozilla) | cut -d '"' -f 2)
-elif [ $(which firefox 2> /dev/null) ] && grep MOZILLA_FIVE_HOME= "$(which firefox)" > /dev/null ; then
-	MOZILLA_HOME=$(grep MOZILLA_FIVE_HOME= $(which firefox) | cut -d '"' -f 2)
+elif [ $(which firefox 2> /dev/null) ] && grep MOZILLA_LIBDIR= "$(which firefox)" > /dev/null ; then
+	MOZILLA_HOME=$(grep MOZILLA_LIBDIR= $(which firefox) | cut -d '=' -f 2)
+elif [ $(which seamonkey 2> /dev/null) ] && grep MOZILLA_LIBDIR= "$(which seamonkey)" > /dev/null ; then
+	MOZILLA_HOME=$(grep MOZILLA_LIBDIR= $(which seamonkey) | cut -d '=' -f 2)
 fi
 
 if [ -n $LD_LIBRARY_PATH ]; then
