diff -aur gtksourceview-sharp-2.0-0.10-orig/configure.in gtksourceview-sharp-2.0-0.10/configure.in
--- gtksourceview-sharp-2.0-0.10-orig/configure.in	2005-05-14 13:19:45.000000000 -0700
+++ gtksourceview-sharp-2.0-0.10/configure.in	2006-01-21 11:16:13.000000000 -0800
@@ -98,18 +98,21 @@
 fi
 AC_SUBST(GAPI_CODEGEN)
 
-MONODOC_REQUIRED_VERSION=1.0
+MONODOC_REQUIRED_VERSION=1.1.8
 PKG_CHECK_MODULES(MONODOC_DEPENDENCY, monodoc >= $MONODOC_REQUIRED_VERSION, enable_monodoc=yes, enable_monodoc=no)
                                                                                 
 if test "x$enable_monodoc" = "xyes"; then 
-    AC_PATH_PROG(MONODOC, monodoc, no)
-    if test "x$MONODOC" = "xno"; then
+    AC_PATH_PROG(MONODOCER, monodocer, no)
+    AC_PATH_PROG(MDASSEMBLER, mdassembler, no)
+    if test "x$MONODOCER" = "xno" -o "x$MDASSEMBLER" = "xno"; then
        enable_monodoc=no
     fi
 else
-    MONODOC=
+    MONODOCER=
+    MDASSEMBLER=
 fi
-AC_SUBST(MONODOC)
+AC_SUBST(MONODOCER)
+AC_SUBST(MDASSEMBLER)
 
 API_VERSION="1.0.0.2"
 AC_SUBST(API_VERSION)
diff -aur gtksourceview-sharp-2.0-0.10-orig/doc/Makefile.am gtksourceview-sharp-2.0-0.10/doc/Makefile.am
--- gtksourceview-sharp-2.0-0.10-orig/doc/Makefile.am	2004-06-06 16:01:25.000000000 -0700
+++ gtksourceview-sharp-2.0-0.10/doc/Makefile.am	2006-01-21 11:36:00.000000000 -0800
@@ -1,6 +1,6 @@
 if ENABLE_MONODOC
 TARGETS = gtksourceview-sharp-docs.zip gtksourceview-sharp-docs.tree gtksourceview-sharp-docs.source
-docsdir = `$(MONODOC) --get-sourcesdir`
+docsdir = `$(PKG_CONFIG) --variable=sourcesdir monodoc`
 docs_DATA = $(TARGETS)
 else
 TARGETS = 
@@ -9,14 +9,11 @@
 endif
 
 
-ASSEMBLER = $(MONODOC) --assemble
-UPDATER = $(MONODOC) --update
-
 gtksourceview-sharp-docs.zip gtksourceview-sharp-docs.tree: $(srcdir)/en/*/*.xml $(srcdir)/en/*.xml
-	$(ASSEMBLER) --ecma $(srcdir)/en -o gtksourceview-sharp-docs
+	$(MDASSEMBLER) --ecma $(srcdir)/en -o gtksourceview-sharp-docs
 
 update:
-		$(UPDATER) ../gtksourceview/gtksourceview-sharp.dll -o ./en -f
+		$(MONODOCER) -assembly:../gtksourceview/gtksourceview-sharp.dll -path:en -updateto:en -delete
 
 CLEANFILES = gtksourceview-sharp-docs.zip gtksourceview-sharp-docs.tree
 
