--- src/addins/CBinding/Navigation/ProjectNodeBuilderExtension.cs	2007-12-20 20:05:02.000000000 +0100
+++ src/addins/CBinding/Navigation/ProjectNodeBuilderExtension.cs.new	2007-12-27 23:27:41.000000000 +0100
@@ -96,7 +96,7 @@
 			check_ctags = true;
 			
 			try {
-				ProcessWrapper p = Runtime.ProcessService.StartProcess ("ctags", "--version", null, null);
+				ProcessWrapper p = Runtime.ProcessService.StartProcess ("exuberant-ctags", "--version", null, null);
 				p.WaitForOutput ();
 				have_ctags = true;
 			} catch {
--- src/addins/CBinding/Parser/TagDatabaseManager.cs	2007-12-27 23:27:54.000000000 +0100
+++ src/addins/CBinding/Parser/TagDatabaseManager.cs.new	2007-12-27 23:28:27.000000000 +0100
@@ -84,7 +84,7 @@
 				args.AppendFormat (" {0}", file.Name);
 			
 			try {
-				ProcessWrapper p = Runtime.ProcessService.StartProcess ("ctags", args.ToString (), tagsDir, null);
+				ProcessWrapper p = Runtime.ProcessService.StartProcess ("exuberant-ctags", args.ToString (), tagsDir, null);
 				p.WaitForExit ();
 			} catch (Exception ex) {
 				throw new IOException ("Could not create tags database (You must have exuberant ctags installed).", ex);
@@ -177,7 +177,7 @@
 				ProcessWrapper p;
 				
 				try {
-					p = Runtime.ProcessService.StartProcess ("ctags", ctags_options, null, null);
+					p = Runtime.ProcessService.StartProcess ("exuberant-ctags", ctags_options, null, null);
 					p.WaitForExit ();
 				} catch (Exception ex) {
 					throw new IOException ("Could not create tags database (You must have exuberant ctags installed).", ex);
@@ -247,7 +247,7 @@
 			ProcessWrapper p;
 			
 			try {
-				p = Runtime.ProcessService.StartProcess ("ctags", ctags_options, null, null);
+				p = Runtime.ProcessService.StartProcess ("exuberant-ctags", ctags_options, null, null);
 				p.WaitForExit (10000);	//If no return detected in 10s, kill anyway
 			} catch (Exception ex) {
 				throw new IOException ("Could not create tags database (You must have exuberant ctags installed).", ex);
