--- configure.in	2007-12-18 20:11:34.000000000 +0100
+++ configure.in.new	2007-12-20 11:08:29.000000000 +0100
@@ -29,9 +29,6 @@
 
 if test "x$enable_update_mimedb" = "xyes"; then
   AC_PATH_PROG(UPDATE_MIME_DB, update-mime-database, no)
-  if test "x$UPDATE_MIME_DB" = "xno"; then
-    AC_MSG_ERROR([You need to install update-mime-database])
-  fi
 fi
 AC_SUBST(UPDATE_MIME_DB)
 
@@ -43,9 +40,6 @@
 
 if test "x$enable_update_desktopdb" = "xyes"; then
   AC_PATH_PROG(UPDATE_DESKTOP_DB, update-desktop-database, no)
-  if test "x$UPDATE_DESKTOP_DB" = "xno"; then
-    AC_MSG_ERROR([You need to install update-desktop-database])
-  fi
 fi
 AC_SUBST(UPDATE_DESKTOP_DB)
 
@@ -245,8 +239,9 @@
 	
 	mozilla_script=$(which mozilla 2> /dev/null)
 	firefox_script=$(which firefox 2> /dev/null)
+	seamonkey_script=$(which seamonkey 2> /dev/null)
 	
-	for runtime_script in "$firefox_script $mozilla_script"; do
+	for runtime_script in "$firefox_script $mozilla_script $seamonkey_script"; do
 		MOZDIR=$(grep "MOZ_DIST_LIB=" $runtime_script 2> /dev/null | cut -d '"' -f 2 | cut -d '=' -f 2)
 		if test -e "$MOZDIR/$MOZFILE"; then echo $MOZDIR; return; fi
 		MOZDIR=$(grep "MOZILLA_FIVE_HOME=" $runtime_script 2> /dev/null | cut -d '"' -f 2 | cut -d '=' -f 2)
