--- src/addins/MonoDevelop.GtkCore/Makefile.am	2008-03-11 03:19:56.000000000 +0100
+++ src/addins/MonoDevelop.GtkCore/Makefile.am.new	2008-03-20 14:46:21.000000000 +0100
@@ -84,7 +84,9 @@
 	svn cp -m "Update Stetic sources" svn+ssh://mono-cvs.ximian.com/source/trunk/stetic svn+ssh://mono-cvs.ximian.com/source/trunk/monodevelop/main/src/addins/MonoDevelop.GtkCore/lib/stetic
 
 if HAVE_GNOME_SHARP
-all: $(ASSEMBLY)
+all: stetic_dlls $(ASSEMBLY)
+stetic_dlls:
+	cd lib ; make -f Makefile ; cd ..
 else
 all:
 endif
@@ -98,6 +100,9 @@
 	$(CSC) $(CSC_FLAGS) -debug -out:$@ -target:library \
 	$(build_resources:%=/resource:%) $(build_sources) $(REFS)
 
+$(ASSEMBLY).mdb: $(ASSEMBLY)
+	true
+
 if HAVE_GNOME_SHARP
 assemblydir = $(MD_ADDIN_DIR)/MonoDevelop.GtkCore
 assembly_DATA = $(ASSEMBLY) $(ASSEMBLY).mdb

--- src/addins/MonoDevelop.WebReferences/Makefile.am	2008-03-11 03:20:16.000000000 +0100
+++ src/addins/MonoDevelop.WebReferences/Makefile.am.new	2008-03-20 14:44:25.000000000 +0100
@@ -56,6 +56,10 @@
 	$(CSC) $(CSC_FLAGS) -debug -out:$@ -target:library \
 	$(build_resources:%=/resource:%) $(build_sources) $(REFS)
 
+$(ASSEMBLY).mdb: $(ASSEMBLY)
+	true
+
+
 assemblydir = $(MD_ADDIN_DIR)/MonoDevelop.WebReferences
 assembly_DATA = $(ASSEMBLY) $(ASSEMBLY).mdb
 
