--- src/addins/CBinding/Navigation/ProjectNodeBuilderExtension.cs	2007-12-20 20:05:02.000000000 +0100
+++ src/addins/CBinding/Navigation/ProjectNodeBuilderExtension.cs	2007-12-27 23:27:41.000000000 +0100
@@ -96,7 +96,7 @@
 			check_ctags = true;
 			
 			try {
-				ProcessWrapper p = Runtime.ProcessService.StartProcess ("ctags", "--version", null, null);
+				ProcessWrapper p = Runtime.ProcessService.StartProcess ("exuberant-ctags", "--version", null, null);
 				p.WaitForOutput ();
 				have_ctags = true;
 			} catch {
--- src/addins/CBinding/Parser/TagDatabaseManager.cs	2008-03-01 15:03:07.000000000 -0800
+++ src/addins/CBinding/Parser/TagDatabaseManager.cs	2008-03-01 15:03:53.000000000 -0800
@@ -77,7 +77,7 @@
 				if (!checkedCtagsInstalled) {
 					checkedCtagsInstalled = true;
 					try {
-						Runtime.ProcessService.StartProcess ("ctags", "--version", null, null).WaitForOutput ();
+						Runtime.ProcessService.StartProcess ("exuberant-ctags", "--version", null, null).WaitForOutput ();
 					} catch {
 						LoggingService.LogWarning ("Cannot update C/C++ tags database because exuberant ctags is not installed.");
 						return false;
@@ -183,7 +183,7 @@
 				ProcessWrapper p;
 				
 				try {
-					p = Runtime.ProcessService.StartProcess ("ctags", ctags_options, null, null);
+					p = Runtime.ProcessService.StartProcess ("exuberant-ctags", ctags_options, null, null);
 					p.WaitForExit ();
 				} catch (Exception ex) {
 					throw new IOException ("Could not create tags database (You must have exuberant ctags installed).", ex);
@@ -259,7 +259,7 @@
 			ProcessWrapper p;
 			
 			try {
-				p = Runtime.ProcessService.StartProcess ("ctags", ctags_options, null, null);
+				p = Runtime.ProcessService.StartProcess ("exuberant-ctags", ctags_options, null, null);
 				p.WaitForExit (10000);	//If no return detected in 10s, kill anyway
 			} catch (Exception ex) {
 				throw new IOException ("Could not create tags database (You must have exuberant ctags installed).", ex);
