--- configure.in	2008-03-11 03:22:00.000000000 +0100
+++ configure.in.new	2008-03-17 15:58:08.000000000 +0100
@@ -244,8 +244,9 @@
 	
 	mozilla_script=$(which mozilla 2> /dev/null)
 	firefox_script=$(which firefox 2> /dev/null)
+	seamonkey_script=$(which seamonkey 2> /dev/null)
 	
-	for runtime_script in "$firefox_script $mozilla_script"; do
+	for runtime_script in "$firefox_script $mozilla_script $seamonkey_script"; do
 		MOZDIR=$(grep "MOZ_DIST_LIB=" $runtime_script 2> /dev/null | cut -d '"' -f 2 | cut -d '=' -f 2)
 		if test -e "$MOZDIR/$MOZFILE"; then echo $MOZDIR; return; fi
 		MOZDIR=$(grep "MOZILLA_FIVE_HOME=" $runtime_script 2> /dev/null | cut -d '"' -f 2 | cut -d '=' -f 2)
