--- 010.Irony.2012.csproj	2016-01-06 12:12:51.564414219 +0300
+++ 010.Irony.2012.csproj.old	2015-03-28 01:21:49.000000000 +0300
@@ -3,7 +3,7 @@
   <PropertyGroup>
     <Configuration Condition=" '$(Configuration)' == '' ">Debug</Configuration>
     <Platform Condition=" '$(Platform)' == '' ">AnyCPU</Platform>
-    <ProductVersion>8.0.30703</ProductVersion>
+    <ProductVersion>9.0.30729</ProductVersion>
     <SchemaVersion>2.0</SchemaVersion>
     <ProjectGuid>{D81F5C91-D7DB-46E5-BC99-49488FB6814C}</ProjectGuid>
     <OutputType>Library</OutputType>
@@ -60,6 +60,8 @@
     <DebugType>pdbonly</DebugType>
     <Optimize>true</Optimize>
     <OutputPath>bin\Release\</OutputPath>
+    <DefineConstants>
+    </DefineConstants>
     <ErrorReport>prompt</ErrorReport>
     <WarningLevel>4</WarningLevel>
     <CodeAnalysisRuleSet>AllRules.ruleset</CodeAnalysisRuleSet>
@@ -67,7 +69,9 @@
   </PropertyGroup>
   <ItemGroup>
     <Reference Include="System" />
-    <Reference Include="System.Core" />
+    <Reference Include="System.Core">
+      <RequiredTargetFramework>3.5</RequiredTargetFramework>
+    </Reference>
     <Reference Include="System.Data" />
     <Reference Include="System.Drawing" />
     <Reference Include="System.Numerics" />
@@ -106,6 +110,7 @@
     <Compile Include="Parsing\Terminals\ImpliedSymbolTerminal.cs" />
     <Compile Include="Parsing\Terminals\LineContinuationTerminal.cs" />
     <Compile Include="Parsing\Terminals\QuotedValueLiteral.cs" />
+    <Compile Include="Parsing\Terminals\RegexLiteral.cs" />
     <Compile Include="Parsing\Terminals\FreeTextLiteral.cs" />
     <Compile Include="Parsing\Terminals\WikiTerminals\WikiBlockTerminal.cs" />
     <Compile Include="Parsing\Terminals\WikiTerminals\WikiTagTerminal.cs" />
@@ -146,6 +151,7 @@
     <Compile Include="Parsing\Terminals\_Terminal.cs" />
     <Compile Include="Parsing\Scanner\Token.cs" />
     <Compile Include="Parsing\Scanner\Scanner.cs" />
+    <Compile Include="Parsing\Terminals\RegexBasedTerminal.cs" />
     <Compile Include="Parsing\Grammar\BnfTerm.cs" />
     <Compile Include="Parsing\Grammar\Grammar.cs" />
     <Compile Include="Parsing\Terminals\TerminalFactory.cs" />
@@ -157,8 +163,6 @@
       <DesignTime>True</DesignTime>
       <DependentUpon>Resources.resx</DependentUpon>
     </Compile>
-    <Compile Include="Parsing\Terminals\RegExLiteral.cs" />
-    <Compile Include="Parsing\Terminals\RegExBasedTerminal.cs" />
   </ItemGroup>
   <ItemGroup>
     <EmbeddedResource Include="Resources.resx">
