diff -NrU5 evolution-sharp-0.18.1.orig/evolution/src/CalComponent.cs evolution-sharp-0.18.1/evolution/src/CalComponent.cs
--- evolution-sharp-0.18.1.orig/evolution/src/CalComponent.cs	2008-11-27 01:42:25.000000000 +0100
+++ evolution-sharp-0.18.1/evolution/src/CalComponent.cs	2008-11-27 01:43:09.000000000 +0100
@@ -992,8 +992,8 @@
 	    {
 		int ret = e_cal_component_get_x_prop_as_int (Handle, name);
 
 		return ret;
 	    }
-	}
 #endif
+	}
 }
