Index: evolution-sharp-0.18.0/evolution/Makefile.am
===================================================================
--- evolution-sharp-0.18.0.orig/evolution/Makefile.am
+++ evolution-sharp-0.18.0/evolution/Makefile.am
@@ -92,14 +92,14 @@ all: $(ASSEMBLY) $(TEST_TARGETS)
 
 install-data-local:
 	@if test -n '$(TARGET)'; then									\
-	  echo "$(GACUTIL) /i $(ASSEMBLY) /f /package $(PACKAGE) /root $(DESTDIR)$(prefix)/lib";		\
-	  $(GACUTIL) /i $(ASSEMBLY) /f /package $(PACKAGE) /root $(DESTDIR)$(prefix)/lib || exit 1;		\
+	  echo "$(GACUTIL) /i $(ASSEMBLY) /f /package $(PACKAGE) /root $(DESTDIR)$(libdir)";		\
+	  $(GACUTIL) /i $(ASSEMBLY) /f /package $(PACKAGE) /root $(DESTDIR)$(libdir) || exit 1;		\
 	fi
 
 uninstall-local:
 	@if test -n '$(TARGET)'; then									\
-	  echo "$(GACUTIL) /u $(ASSEMBLY_NAME) /package $(PACKAGE) /root $(DESTDIR)$(prefix)/lib";		\
-	  $(GACUTIL) /u $(ASSEMBLY_NAME) /package $(PACKAGE) /root $(DESTDIR)$(prefix)/lib || exit 1;	\
+	  echo "$(GACUTIL) /u $(ASSEMBLY_NAME) /package $(PACKAGE) /root $(DESTDIR)$(libdir)";		\
+	  $(GACUTIL) /u $(ASSEMBLY_NAME) /package $(PACKAGE) /root $(DESTDIR)$(libdir) || exit 1;	\
 	fi
 
 evolution-sharp.snk: $(top_srcdir)/evolution-sharp.snk
Index: evolution-sharp-0.18.0/evolution-sharp.pc.in
===================================================================
--- evolution-sharp-0.18.0.orig/evolution-sharp.pc.in
+++ evolution-sharp-0.18.0/evolution-sharp.pc.in
@@ -1,5 +1,6 @@
 prefix=@prefix@
-libdir=${prefix}/lib
+exec_prefix=@exec_prefix@
+libdir=@libdir@
 
 Name: Evolution#
 Description: Evolution# - Evolution .NET Binding
