diff --git a/soup-sharp-2.4.pc.in b/soup-sharp-2.4.pc.in
index c2dd575..84e5e2c 100644
--- a/soup-sharp-2.4.pc.in
+++ b/soup-sharp-2.4.pc.in
@@ -1,9 +1,10 @@
 prefix=@prefix@
 exec_prefix=${prefix}
 libdir=@libdir@
+pkglibdir=${exec_prefix}/lib/mono/@ASSEMBLY_NAME@
 
 Name: @ASSEMBLY_TITLE@
 Description: @ASSEMBLY_DESCRIPTION@
 Version: @VERSION@
 Requires: 
-Libs: -r:${libdir}/mono/@ASSEMBLY_NAME@/@ASSEMBLY_NAME@.dll
+Libs: -r:${pkglibdir}/@ASSEMBLY_NAME@.dll
