--- beagle-0.3.8old/search/Beagle.Search.Tiles/Tile.cs	2008-06-12 07:41:19.000000000 +0200
+++ beagle-0.3.8/search/Beagle.Search.Tiles/Tile.cs	2008-10-18 20:25:23.000000000 +0200
@@ -348,7 +348,7 @@
 #else
 			MimeApplication app;
 			app = Mime.GetDefaultApplication (mimetype);
-			bool expect_uris = app.SupportUris ();
+			bool expect_uris = app.SupportsUris ();
 			path = hit.Path;
 
 			if (app == null) {
