Index: beagled/BeagleDaemon.cs
===================================================================
RCS file: /cvs/gnome/beagle/beagled/BeagleDaemon.cs,v
retrieving revision 1.109
diff -u -p -u -r1.109 BeagleDaemon.cs
--- beagled/BeagleDaemon.cs	8 Mar 2006 21:01:50 -0000	1.109
+++ beagled/BeagleDaemon.cs	17 Mar 2006 18:27:58 -0000
@@ -371,10 +371,7 @@ namespace Beagle.Daemon {
 
 			Log.Initialize (PathFinder.LogDir,
 					"Beagle", 
-					// FIXME: We always turn on full debugging output!  We are still
-					// debugging this code, after all...
-					//arg_debug ? LogLevel.Debug : LogLevel.Warn,
-					LogLevel.Debug,
+					arg_debug ? LogLevel.Debug : LogLevel.Warn,
 					arg_fg);
 
 			Logger.Log.Info ("Starting Beagle Daemon (version {0})", ExternalStringsHack.Version);
Index: beagled/IndexHelper/IndexHelper.cs
===================================================================
RCS file: /cvs/gnome/beagle/beagled/IndexHelper/IndexHelper.cs,v
retrieving revision 1.24
diff -u -p -u -r1.24 IndexHelper.cs
--- beagled/IndexHelper/IndexHelper.cs	8 Feb 2006 18:19:49 -0000	1.24
+++ beagled/IndexHelper/IndexHelper.cs	4 Apr 2006 20:08:28 -0000
@@ -61,16 +61,13 @@ namespace Beagle.IndexHelper {
 			bool run_by_hand = (Environment.GetEnvironmentVariable ("BEAGLE_RUN_HELPER_BY_HAND") != null);
 			bool log_in_fg = (Environment.GetEnvironmentVariable ("BEAGLE_LOG_IN_THE_FOREGROUND_PLEASE") != null);
 			
-			// FIXME: We always turn on full debugging output!  We are still
-			// debugging this code, after all...
-			//bool debug = (Environment.GetEnvironmentVariable ("BEAGLE_DEBUG_FLAG_IS_SET") != null);
+			bool debug = (Environment.GetEnvironmentVariable ("BEAGLE_DEBUG_FLAG_IS_SET") != null);
 
 			last_activity = DateTime.Now;
 
 			Log.Initialize (PathFinder.LogDir,
 					"IndexHelper",
-					//debug ? LogLevel.Debug : LogLevel.Warn,
-					LogLevel.Debug,
+					debug ? LogLevel.Debug : LogLevel.Warn,
 					run_by_hand || log_in_fg);
 
 			Application.InitCheck ("IndexHelper", ref args);
