diff -ru beagle-0.2.18~/search/UIManager.cs beagle-0.2.18/search/UIManager.cs
--- beagle-0.2.18~/search/UIManager.cs	2008-06-29 15:14:28.000000000 +0200
+++ beagle-0.2.18/search/UIManager.cs	2008-06-29 15:15:00.000000000 +0200
@@ -394,7 +394,7 @@
 			if (ScopeChanged == null)
 				return;
 
-			ScopeType scope = (ScopeType) System.Enum.Parse (typeof (ScopeType), ((Action) obj).Name);
+			ScopeType scope = (ScopeType) System.Enum.Parse (typeof (ScopeType), ((Gtk.Action) obj).Name);
 			
 			ScopeChanged (scope, ((ToggleAction) obj).Active);
 		}
