--- a/MyPad/MyPad.csproj
+++ b/MyPad/MyPad.csproj
@@ -58,6 +58,9 @@
     </Reference>
     <Reference Include="System.Configuration" />
     <Reference Include="System.ServiceModel" />
+    <Reference Include="ICSharpCode.TextEditor">
+      <HintPath>..\..\..\NuGet\ICSharpCode.TextEditor.3.2.2\lib\net40\ICSharpCode.TextEditor</HintPath>
+    </Reference>
   </ItemGroup>
   <ItemGroup>
     <Compile Include="Dialogs\AboutDialog.cs">
@@ -271,10 +210,4 @@
       </Properties>
     </MonoDevelop>
   </ProjectExtensions>
-  <ItemGroup>
-    <ProjectReference Include="..\..\ICSharpCode.TextEditor.git\Project\ICSharpCode.TextEditor.csproj">
-      <Project>{2D18BE89-D210-49EB-A9DD-2246FBB3DF6D}</Project>
-      <Name>ICSharpCode.TextEditor</Name>
-    </ProjectReference>
-  </ItemGroup>
 </Project>
\ No newline at end of file
