--- mono/metadata/threads.h	2007/12/20 00:38:27	1.1
+++ mono/metadata/threads.h	2007/12/20 00:38:46
@@ -14,7 +14,6 @@
 #include <mono/utils/mono-membar.h>
 #include <mono/metadata/object.h>
 #include <mono/metadata/appdomain.h>
-#include <mono/metadata/threads-types.h>
 
 G_BEGIN_DECLS
 
@@ -67,10 +66,6 @@ extern void mono_thread_interruption_che
 extern void mono_thread_force_interruption_checkpoint (void);
 extern gint32* mono_thread_interruption_request_flag (void);
 
-extern void mono_thread_set_state (MonoThread *thread, MonoThreadState state);
-extern void mono_thread_clr_state (MonoThread *thread, MonoThreadState state);
-extern gboolean mono_thread_test_state (MonoThread *thread, MonoThreadState test);
-
 G_END_DECLS
 
 #endif /* _MONO_METADATA_THREADS_H_ */
--- mono/metadata/threads-types.h	2007/12/20 00:38:32	1.1
+++ mono/metadata/threads-types.h	2007/12/20 00:40:55
@@ -157,4 +157,8 @@ extern MonoThreadHazardPointers* mono_ha
 		(hp)->hazard_pointers [(i)] = NULL; \
 	} while (0)
 
+extern void mono_thread_set_state (MonoThread *thread, MonoThreadState state) MONO_INTERNAL;
+extern void mono_thread_clr_state (MonoThread *thread, MonoThreadState state) MONO_INTERNAL;
+extern gboolean mono_thread_test_state (MonoThread *thread, MonoThreadState test) MONO_INTERNAL;
+
 #endif /* _MONO_METADATA_THREADS_TYPES_H_ */
