--- trunk/mcs/mcs/expression.cs	2007/12/22 13:17:22	91807
+++ trunk/mcs/mcs/expression.cs	2007/12/23 13:29:15	91831
@@ -4242,6 +4242,10 @@
 
 		public override Expression DoResolve (EmitContext ec)
 		{
+			// Don't resolve already resolved expression
+			if (eclass != ExprClass.Invalid)
+				return this;
+			
 			Expression expr_resolved = expr.Resolve (ec, ResolveFlags.VariableOrValue | ResolveFlags.MethodGroup);
 			if (expr_resolved == null)
 				return null;
