--- a/src/SolutionInfo.cs
+++ b/src/SolutionInfo.cs
@@ -2,6 +2,6 @@
 using System.Reflection;
 using System.Runtime.CompilerServices;
 using System.Runtime.InteropServices;
-[assembly: AssemblyVersion("1.9.2")]
-[assembly: AssemblyFileVersion("1.9.2")]
-[assembly: AssemblyInformationalVersion("1.9.2.0da451")]
+[assembly: AssemblyVersion("1.9.3")]
+[assembly: AssemblyFileVersion("1.9.3")]
+[assembly: AssemblyInformationalVersion("1.9.3.0")]
