http://bugzilla.gnome.org/show_bug.cgi?id=566355

diff -NrU5 tasque-0.1.8.orig/configure.ac tasque-0.1.8/configure.ac
--- tasque-0.1.8.orig/configure.ac	2009-01-09 18:44:30.000000000 +0100
+++ tasque-0.1.8/configure.ac	2009-01-09 19:19:06.000000000 +0100
@@ -53,39 +53,12 @@
 AC_SUBST(NOTIFY_SHARP_LIBS)
 
 AC_ARG_ENABLE(debug,
 	AC_HELP_STRING([--enable-debug],
 		[Use 'Debug' Configuration [default=yes]]),
-		enable_debug=$enableval, enable_debug=no)
+		enable_debug=$enableval, enable_debug=yes)
 AM_CONDITIONAL(ENABLE_DEBUG, test "x$enable_debug" = "xyes")
-if test "x$enable_debug" = "xyes" ; then
-#	DEBUG_CONFIG_LIBRARIES=' ${pkglibdir}/Db4objects.Db4o.dll'
-#	DEBUG_CONFIG_LIBS=' -r:${pkglibdir}/Db4objects.Db4o.dll'
-#	AC_SUBST(DEBUG_CONFIG_LIBRARIES)
-#	AC_SUBST(DEBUG_CONFIG_LIBS)
-	CONFIG_REQUESTED="yes"
-fi
-AC_ARG_ENABLE(release,
-	AC_HELP_STRING([--enable-release],
-		[Use 'Release' Configuration [default=no]]),
-		enable_release=$enableval, enable_release=no)
-AM_CONDITIONAL(ENABLE_RELEASE, test "x$enable_release" = "xyes")
-if test "x$enable_release" = "xyes" ; then
-#	RELEASE_CONFIG_LIBRARIES=' ${pkglibdir}/Db4objects.Db4o.dll'
-#	RELEASE_CONFIG_LIBS=' -r:${pkglibdir}/Db4objects.Db4o.dll'
-#	AC_SUBST(RELEASE_CONFIG_LIBRARIES)
-#	AC_SUBST(RELEASE_CONFIG_LIBS)
-	CONFIG_REQUESTED="yes"
-fi
-if test -z "$CONFIG_REQUESTED" ; then
-#	DEBUG_CONFIG_LIBRARIES=' ${pkglibdir}/Db4objects.Db4o.dll'
-#	DEBUG_CONFIG_LIBS=' -r:${pkglibdir}/Db4objects.Db4o.dll'
-#	AC_SUBST(DEBUG_CONFIG_LIBRARIES)
-#	AC_SUBST(DEBUG_CONFIG_LIBS)
-	AM_CONDITIONAL(ENABLE_DEBUG, true)
-fi
-
 
 #
 # DEFINES uses for #if statements
 #
 AC_ARG_WITH(dbus_service_dir, [  --with-dbus-service-dir=DIR            Where to install Tasque's DBus service file.])
@@ -225,11 +198,10 @@
 echo "
 Configuration:
 
 	Prefix:            ${prefix}
 	Debug build:       ${enable_debug}
-	Release build:     ${enable_release}
 	Notification:      ${enable_notify_sharp}
 	Dummy Backend:     ${enable_backend_dummy}
 	Evolution Backend: ${enable_backend_eds}
 	ICECore Backend:   ${enable_backend_icecore}
 	RTM Backend:       ${enable_backend_rtm}
diff -NrU5 tasque-0.1.8.orig/RtmNet/Makefile.am tasque-0.1.8/RtmNet/Makefile.am
--- tasque-0.1.8.orig/RtmNet/Makefile.am	2009-01-09 18:44:30.000000000 +0100
+++ tasque-0.1.8/RtmNet/Makefile.am	2009-01-09 19:26:42.000000000 +0100
@@ -1,16 +1,18 @@
 CSC = $(GMCS)
 
 TARGET = RtmNet.dll
 
+CSFLAGS = -t:library -noconfig -codepage:utf8
+
 if ENABLE_DEBUG
-CSFLAGS =  -t:library -noconfig -codepage:utf8 -warn:4 -debug -d:DEBUG
-endif
-if ENABLE_RELEASE
-CSFLAGS =  -t:library -noconfig -codepage:utf8 -warn:4
+CSFLAGS += -warn:4 -debug -d:DEBUG
+TARGET_DEBUG_FILE = $(TARGET).mdb
+$(TARGET).mdb: $(TARGET)
 endif
 
+
 CSFILES = \
 	$(srcdir)/ApiKeyRequiredException.cs		\
 	$(srcdir)/AssemblyInfo.cs			\
 	$(srcdir)/Auth.cs				\
 	$(srcdir)/AuthenticationRequiredException.cs	\
@@ -45,20 +47,20 @@
 
 $(TARGET): $(CSFILES)
 	$(CSC) -unsafe -out:$@ $(CSFLAGS) $^ $(ASSEMBLIES) $(RESOURCES)
 
 tasquelibdir = $(prefix)/lib/tasque
-tasquelib_DATA = $(TARGET)	
+tasquelib_DATA = $(TARGET) $(TARGET_DEBUG_FILE)
 
 bin_SCRIPTS = $(WRAPPER)
 
 
 EXTRA_DIST = \
 	$(CSFILES)
 
 CLEANFILES = \
-	$(TARGET)					\
-	$(TARGET).mdb
+	$(TARGET_DEBUG_FILE)		\
+	$(TARGET)
 
 DISTCLEANFILES =                        \
-	$(TARGET)			\
-	$(TARGET).mdb
+	$(TARGET_DEBUG_FILE)		\
+	$(TARGET)
diff -NrU5 tasque-0.1.8.orig/src/Makefile.am tasque-0.1.8/src/Makefile.am
--- tasque-0.1.8.orig/src/Makefile.am	2009-01-09 18:44:30.000000000 +0100
+++ tasque-0.1.8/src/Makefile.am	2009-01-09 19:15:58.000000000 +0100
@@ -2,20 +2,21 @@
 
 CSC = $(GMCS)
 
 TARGET = Tasque.exe
 WRAPPER = tasque
+CSFLAGS = -noconfig -codepage:utf8
 
 if GTK_2_12
 GTK_2_12_CSFLAGS = -define:GTK_2_12
+CSFLAGS += $(GTK_2_12_CSFLAGS)
 endif
 
 if ENABLE_DEBUG
-CSFLAGS =  -noconfig -codepage:utf8 -warn:4 -debug -d:DEBUG $(GTK_2_12_CSFLAGS)
-endif
-if ENABLE_RELEASE
-CSFLAGS =  -noconfig -codepage:utf8 -warn:4 $(GTK_2_12_CSFLAGS)
+CSFLAGS += -warn:4 -debug -d:DEBUG
+TARGET_DEBUG_FILE = $(TARGET).mdb
+$(TARGET).mdb: $(TARGET)
 endif
 
 if ENABLE_NOTIFY_SHARP
 NOTIFY_SHARP_CSFLAGS = -define:ENABLE_NOTIFY_SHARP
 endif
@@ -138,11 +139,11 @@
 
 $(TARGET): $(CSFILES) Defines.cs
 	$(CSC) -unsafe -out:$@ $(CSFLAGS) $(NOTIFY_SHARP_CSFLAGS) $^ $(ASSEMBLIES) $(RESOURCES)
 
 tasquelibdir = $(libdir)/tasque
-tasquelib_DATA = $(TARGET) $(TARGET).config $(TARGET).mdb
+tasquelib_DATA = $(TARGET) $(TARGET).config $(TARGET_DEBUG_FILE)
 
 bin_SCRIPTS = $(WRAPPER)
 
 $(WRAPPER): $(srcdir)/$(WRAPPER).in Makefile
 	sed -e "s|\@prefix\@|$(prefix)|g"               \
@@ -154,12 +155,10 @@
 		-e "s|\@wrapper\@|$(WRAPPER)|g"             \
 		-e "s|\@srcdir\@|$(PWD)|g"  \
 		< $< > $@
 	chmod +x $(WRAPPER)
 
-$(TARGET).mdb: $(TARGET)
-
 $(TARGET).config: $(srcdir)/$(TARGET).config.in Makefile
 	sed -e "s|\@pkglibdir\@|$(pkglibdir)|" \
 	    < $< > $@
 
 Defines.cs: $(srcdir)/Defines.cs.in Makefile
@@ -187,15 +186,15 @@
 pkgconfig_DATA = tasque.pc
 
 CLEANFILES = \
 	$(TARGET)					\
 	$(TARGET).config	\
-	$(TARGET).mdb				\
+	$(TARGET_DEBUG_FILE)				\
 	$(WRAPPER)					\
 	Defines.cs
 
 DISTCLEANFILES =                        \
         $(WRAPPER)			\
 	$(TARGET)			\
 	$(TARGET).config	\
-	$(TARGET).mdb		\
+	$(TARGET_DEBUG_FILE)		\
 	Defines.cs
