--- before/corefx-1.1.1/src/Native/Unix/CMakeLists.txt	2017-02-22 19:42:11.000000000 +0300
+++ after/corefx-1.1.1/src/Native/Unix/CMakeLists.txt	2017-07-01 00:59:39.256957823 +0300
@@ -15,7 +15,7 @@
 add_compile_options(-Wno-disabled-macro-expansion)
 add_compile_options(-Wno-c++98-compat)
 add_compile_options(-Wno-c++98-compat-pedantic)
-add_compile_options(-Werror)
+#add_compile_options(-Werror)
 add_compile_options(-fPIC)
 add_compile_options(-I${CMAKE_CURRENT_SOURCE_DIR}/Common)
 add_compile_options(-I${CMAKE_CURRENT_BINARY_DIR}/Common)
