diff --git a/coreclr-1.0.6/src/debug/daccess/dacdbiimpl.cpp b/coreclr-1.0.6/src/debug/daccess/dacdbiimpl.cpp
index 26e3d6c..59f217c 100644
--- a/coreclr-1.0.6/src/debug/daccess/dacdbiimpl.cpp
+++ b/coreclr-1.0.6/src/debug/daccess/dacdbiimpl.cpp
@@ -90,7 +90,6 @@ IDacDbiInterface::IAllocator * g_pAllocator = NULL;
 //
 
 // Need a class to serve as a tag that we can use to overload New/Delete.
-#define forDbi (*(forDbiWorker *)NULL)
 
 void * operator new(size_t lenBytes, const forDbiWorker &)
 {
diff --git a/coreclr-1.0.6/src/debug/ee/debugger.h b/coreclr-1.0.6/src/debug/ee/debugger.h
index 6368647..59d1e66 100644
--- a/coreclr-1.0.6/src/debug/ee/debugger.h
+++ b/coreclr-1.0.6/src/debug/ee/debugger.h
@@ -3512,10 +3512,10 @@ class DebuggerEval
  * ------------------------------------------------------------------------ */
 
 class InteropSafe {};
-#define interopsafe (*(InteropSafe*)NULL)
+SELECTANY InteropSafe interopsafe;
 
 class InteropSafeExecutable {};
-#define interopsafeEXEC (*(InteropSafeExecutable*)NULL)
+SELECTANY InteropSafeExecutable interopsafeEXEC;
 
 #ifndef DACCESS_COMPILE
 inline void * __cdecl operator new(size_t n, const InteropSafe&)
diff --git a/coreclr-1.0.6/src/debug/ildbsymlib/symwrite.h b/coreclr-1.0.6/src/debug/ildbsymlib/symwrite.h
index 055b8ec..54ab11a 100644
--- a/coreclr-1.0.6/src/debug/ildbsymlib/symwrite.h
+++ b/coreclr-1.0.6/src/debug/ildbsymlib/symwrite.h
@@ -839,7 +839,8 @@ class ArrayStorage
         {
             // Help mitigate the impact of buffer overflow
             // Fail fast with a null-reference AV
-            return *(static_cast<T*>(0)) ;
+            volatile char* nullPointer = nullptr;
+            *nullPointer;
         }
         return m_array[ i ];
     }
diff --git a/coreclr-1.0.6/src/debug/inc/dacdbiinterface.h b/coreclr-1.0.6/src/debug/inc/dacdbiinterface.h
index e61e240..24c3e24 100644
--- a/coreclr-1.0.6/src/debug/inc/dacdbiinterface.h
+++ b/coreclr-1.0.6/src/debug/inc/dacdbiinterface.h
@@ -32,7 +32,7 @@
 template<class T> void DeleteDbiMemory(T *p);
 // Need a class to serve as a tag that we can use to overload New/Delete.
 class forDbiWorker {};
-#define forDbi (*(forDbiWorker *)NULL)
+SELECTANY forDbiWorker forDbi;
 extern void * operator new(size_t lenBytes, const forDbiWorker &);
 extern void * operator new[](size_t lenBytes, const forDbiWorker &);
 extern void operator delete(void *p, const forDbiWorker &);
diff --git a/coreclr-1.0.6/src/pal/src/exception/seh.cpp b/coreclr-1.0.6/src/pal/src/exception/seh.cpp
index 473c490..ad09e02 100644
--- a/coreclr-1.0.6/src/pal/src/exception/seh.cpp
+++ b/coreclr-1.0.6/src/pal/src/exception/seh.cpp
@@ -274,7 +274,7 @@ SEHProcessException(PAL_SEHException* exception)
                     {
                         // The exception happened in the page right below the stack limit,
                         // so it is a stack overflow
-                        write(STDERR_FILENO, StackOverflowMessage, sizeof(StackOverflowMessage) - 1);
+                        (void)write(STDERR_FILENO, StackOverflowMessage, sizeof(StackOverflowMessage) - 1);
                         PROCAbort();
                     }
                 }
diff --git a/coreclr-1.0.6/tests/src/Common/Platform/platformdefines.cpp b/coreclr-1.0.6/tests/src/Common/Platform/platformdefines.cpp
index 4bef170..82061ac 100644
--- a/coreclr-1.0.6/tests/src/Common/Platform/platformdefines.cpp
+++ b/coreclr-1.0.6/tests/src/Common/Platform/platformdefines.cpp
@@ -277,7 +277,7 @@ DWORD TP_GetFullPathName(LPWSTR fileName, DWORD nBufferLength, LPWSTR lpBuffer)
 	return GetFullPathNameW(fileName, nBufferLength, lpBuffer, NULL);
 #else
 	char nativeFullPath[MAX_PATH];
-	realpath(HackyConvertToSTR(fileName), nativeFullPath);
+	(void)realpath(HackyConvertToSTR(fileName), nativeFullPath);
 	LPWSTR fullPathForCLR = HackyConvertToWSTR(nativeFullPath);
 	wcscpy_s(lpBuffer, MAX_PATH, fullPathForCLR);
 	return wcslen(lpBuffer);
diff --git a/coreclr-1.0.6/tests/src/Common/Platform/platformdefines.h b/coreclr-1.0.6/tests/src/Common/Platform/platformdefines.h
index 49e8f88..c196b0c 100644
--- a/coreclr-1.0.6/tests/src/Common/Platform/platformdefines.h
+++ b/coreclr-1.0.6/tests/src/Common/Platform/platformdefines.h
@@ -87,7 +87,7 @@ typedef void* HMODULE;
 typedef void* ULONG_PTR;
 typedef unsigned error_t;
 typedef void* LPVOID;
-typedef char BYTE;
+typedef unsigned char BYTE;
 typedef WCHAR OLECHAR;
 #endif
 
diff --git a/coreclr-1.0.6/tests/src/Interop/common/types.h b/coreclr-1.0.6/tests/src/Interop/common/types.h
index 7d7f776..cb59c42 100755
--- a/coreclr-1.0.6/tests/src/Interop/common/types.h
+++ b/coreclr-1.0.6/tests/src/Interop/common/types.h
@@ -28,7 +28,7 @@ typedef void* HMODULE;
 typedef void* ULONG_PTR;
 typedef unsigned error_t;
 typedef void* LPVOID;
-typedef char BYTE;
+typedef unsigned char BYTE;
 typedef WCHAR OLECHAR;
 
 typedef unsigned int UINT_PTR;
@@ -54,4 +54,4 @@ typedef int*  DWORD_PTR;
 #define FALSE 0
 #endif
 
-#endif //_INTEROP_TYPES__H
\ No newline at end of file
+#endif //_INTEROP_TYPES__H
