diff -rupN a/coreclr-1.0.6/src/pal/src/include/pal/palinternal.h b/coreclr-1.0.6/src/pal/src/include/pal/palinternal.h
--- a/coreclr-1.0.6/src/pal/src/include/pal/palinternal.h	2017-04-27 21:06:55.268009100 +0300
+++ b/coreclr-1.0.6/src/pal/src/include/pal/palinternal.h	2017-04-27 21:08:23.705175087 +0300
@@ -567,6 +567,8 @@ function_name() to call the system's imp
 #endif
 #include <ctype.h>
 
+#define _GLIBCXX_INCLUDE_NEXT_C_HEADERS 1
+
 #define _WITH_GETLINE
 #include <stdio.h>
 #include <stdlib.h>
diff -rupN a/coreclr-1.0.6/src/pal/src/include/pal/sharedmemory.h b/coreclr-1.0.6/src/pal/src/include/pal/sharedmemory.h
--- a/coreclr-1.0.6/src/pal/src/include/pal/sharedmemory.h	2017-04-27 21:06:55.268009100 +0300
+++ b/coreclr-1.0.6/src/pal/src/include/pal/sharedmemory.h	2017-04-27 21:07:44.351101225 +0300
@@ -182,6 +182,10 @@ public:
     virtual void Close(bool isAbruptShutdown, bool releaseSharedData)
     {
     }
+
+	virtual ~SharedMemoryProcessDataBase()
+	{
+	}
 };
 
 class SharedMemoryProcessDataHeader
