--- configure.in.orig	2008-04-18 11:20:31.000000000 -0500
+++ configure.in	2008-04-18 11:21:01.000000000 -0500
@@ -66,8 +66,8 @@
 PKG_CHECK_MODULES(FONTCONFIG, fontconfig,
 	[fontconfig_pkgconfig=yes], [fontconfig_pkgconfig=no])
 if test "x$FONTCONFIG" = "xno"; then
-	AC_PATH_PROG(FONTCONFIG-CONFIG, fontconfig-config, no)
-	if test "x$FONTCONFIG-CONFIG" = "xyes"; then
+	AC_PATH_PROG(FONTCONFIG_CONFIG, fontconfig-config, no)
+	if test "x$FONTCONFIG_CONFIG" = "xyes"; then
 		FONTCONFIG_CFLAGS="`fontconfig-config --cflags`"
 		FONTCONFIG_LIBS="`fontconfig-config --libs`"
 	fi
@@ -78,8 +78,8 @@
 PKG_CHECK_MODULES(FREETYPE2, freetype2,
 	[freetype2_pkgconfig=yes], freetype2_pkgconfig=no])
 if test "x$FREETYPE2" = "xno"; then
-	AC_PATH_PROG(FREETYPE-CONFIG, freetype-config, no)
-	if test "x$FREETYPE-CONFIG" = "xyes"; then
+	AC_PATH_PROG(FREETYPE_CONFIG, freetype-config, no)
+	if test "x$FREETYPE_CONFIG" = "xyes"; then
 		FREETYPE2_CFLAGS="`freetype-config --cflags`"
 		FREETYPE2_LIBS="`freetype-config --libs`"
 	fi
