diff -aur libgdiplus-1.1.13-orig/configure.in libgdiplus-1.1.13/configure.in
--- libgdiplus-1.1.13-orig/configure.in	2006-01-05 22:30:08.000000000 -0800
+++ libgdiplus-1.1.13/configure.in	2006-01-14 21:32:53.000000000 -0800
@@ -203,7 +203,7 @@
       AC_MSG_WARN(*** GIF loader will not be built (giflibrary not found) ***))
   fi
 
-if ! test x$GIF != x; then
+if test x$with_libgif != xno && ! test x$GIF != x; then
    AC_CHECK_LIB(ungif, DGifOpenFileName,
       [AC_CHECK_HEADER(gif_lib.h,
 	GIF='ungif'; LIBGIF='-lungif',
