--- a/corefx-1.1.1/init-tools.sh	2017-03-11 23:35:22.604691000 -0600
+++ b/corefx-1.1.1/init-tools.sh	2017-03-11 23:39:11.945691000 -0600
@@ -4,7 +4,7 @@
 __init_tools_log=$__scriptpath/init-tools.log
 __PACKAGES_DIR=$__scriptpath/packages
 __TOOLRUNTIME_DIR=$__scriptpath/Tools
-__DOTNET_PATH=$__TOOLRUNTIME_DIR/dotnetcli
+__DOTNET_PATH=$__scriptpath/../tools-dotnet
 __DOTNET_CMD=$__DOTNET_PATH/dotnet
 if [ -z "$__BUILDTOOLS_SOURCE" ]; then __BUILDTOOLS_SOURCE=https://dotnet.myget.org/F/dotnet-buildtools/api/v3/index.json; fi
 __BUILD_TOOLS_PACKAGE_VERSION=$(cat $__scriptpath/BuildToolsVersion.txt)
@@ -59,8 +59,7 @@
         fi
     fi
 
-    # Cannot determine Linux distribution, assuming Ubuntu 14.04.
-    echo "ubuntu"
+    echo "debian"
     return 0
 }
 
@@ -87,6 +86,7 @@
 fi
 if [ ! -e $__INIT_TOOLS_DONE_MARKER ]; then
     if [ -e $__TOOLRUNTIME_DIR ]; then rm -rf -- $__TOOLRUNTIME_DIR; fi
+    mkdir -p $__TOOLRUNTIME_DIR
     echo "Running: $__scriptpath/init-tools.sh" > $__init_tools_log
     if [ ! -e $__DOTNET_PATH ]; then
         echo "Installing dotnet cli..."
@@ -116,6 +116,8 @@
         if [ ! -e "$__BUILD_TOOLS_PATH/init-tools.sh" ]; then echo "ERROR: Could not restore build tools correctly. See '$__init_tools_log' for more details."1>&2; fi
     fi
 
+    patch -p0 < buildtools.patch
+
     echo "Initializing BuildTools..."
     echo "Running: $__BUILD_TOOLS_PATH/init-tools.sh $__scriptpath $__DOTNET_CMD $__TOOLRUNTIME_DIR" >> $__init_tools_log
     $__BUILD_TOOLS_PATH/init-tools.sh $__scriptpath $__DOTNET_CMD $__TOOLRUNTIME_DIR >> $__init_tools_log
