--- before/coreclr-1.1.1/src/pal/inc/pal.h	2017-02-09 10:16:50.000000000 +0300
+++ after/coreclr-1.1.1/src/pal/inc/pal.h	2017-07-01 01:34:53.949949756 +0300
@@ -5967,6 +5967,12 @@
 }
 #endif
 
+#ifndef __has_builtin
+#define __has_builtin(x) 0
+#endif
+
+#if !__has_builtin(_rotl)
+
 /*++
 Function:
 _rotl
@@ -5985,11 +5991,15 @@
     return retval;
 }
 
+#endif
+
 // On 64 bit unix, make the long an int.
 #ifdef BIT64
 #define _lrotl _rotl
 #endif // BIT64
 
+#if !__has_builtin(_rotr)
+
 /*++
 Function:
 _rotr
@@ -6008,6 +6018,8 @@
     return retval;
 }
 
+#endif
+
 PALIMPORT int __cdecl abs(int);
 #ifndef PAL_STDCPP_COMPAT
 PALIMPORT LONG __cdecl labs(LONG);
