diff --git a/sources/Makefile.am b/sources/Makefile.am
index a726b23..aca6456 100644
--- a/sources/Makefile.am
+++ b/sources/Makefile.am
@@ -42,7 +42,7 @@ generated-stamp: $(API)
 
 $(DLL): $(build_sources) generated-stamp
 	$(CSC) -nowarn:169 -unsafe -target:library $(GTK_SHARP_LIBS) \
-		$(build_sources)  $(MONOCAIRO_LIBS) $(SOUPSHARP_LIBS) generated/*.cs -out:$(DLL)
+		$(build_sources)  $(MONOCAIRO_LIBS) $(SOUPSHARP_LIBS) generated/*.cs generated/*/*.cs -out:$(DLL)
 
 install-data-local:
 	echo "$(GACUTIL) /i $(DLL) /f $(GACUTIL_FLAGS)";  \
diff --git a/webkitgtk-sharp-3.0.pc.in b/webkitgtk-sharp-3.0.pc.in
index c2dd575..84e5e2c 100644
--- a/webkitgtk-sharp-3.0.pc.in
+++ b/webkitgtk-sharp-3.0.pc.in
@@ -1,9 +1,10 @@
 prefix=@prefix@
 exec_prefix=${prefix}
 libdir=@libdir@
+pkglibdir=${exec_prefix}/lib/mono/@ASSEMBLY_NAME@
 
 Name: @ASSEMBLY_TITLE@
 Description: @ASSEMBLY_DESCRIPTION@
 Version: @VERSION@
 Requires: 
-Libs: -r:${libdir}/mono/@ASSEMBLY_NAME@/@ASSEMBLY_NAME@.dll
+Libs: -r:${pkglibdir}/@ASSEMBLY_NAME@.dll
