--- mod_mono-1.1.17-old/src/mod_mono.h	2006-08-30 17:05:42.000000000 +0200
+++ mod_mono-1.1.17/src/mod_mono.h	2006-11-10 01:11:22.000000000 +0100
@@ -158,7 +158,7 @@
 #define DIRECTORY_SEPARATOR	":"
 #endif
 
-#define MODMONO_SERVER_PATH 	MONO_PREFIX "/bin/mod-mono-server"
+#define MODMONO_SERVER_PATH 	MONO_PREFIX "/bin/mod-mono-server2"
 #define WAPIDIR				"/tmp"
 #define DOCUMENT_ROOT		NULL
 #define APPCONFIG_FILE		NULL
