diff -NrU5 monsoon-0.20.orig/Monsoon/Makefile.am monsoon-0.20/Monsoon/Makefile.am
--- monsoon-0.20.orig/Monsoon/Makefile.am	2009-02-05 00:41:06.000000000 +0100
+++ monsoon-0.20/Monsoon/Makefile.am	2009-02-05 00:53:41.000000000 +0100
@@ -1,7 +1,6 @@
-
-EXTRA_DIST = $(NLOG_DLL_SOURCE)
+EXTRA_DIST =
 
 # Warning: This is an automatically generated file, do not edit!
 
 if ENABLE_DEBUG
 ASSEMBLY_COMPILER_COMMAND = gmcs
@@ -19,11 +18,10 @@
 RSS_PNG_SOURCE=icons/rss.png
 LOWEST_PNG_SOURCE=icons/lowest.png
 MONSOON_DESKTOP_SOURCE=monsoon.desktop
 DONOTDOWNLOAD_PNG_SOURCE=icons/donotdownload.png
 NORMAL_PNG_SOURCE=icons/normal.png
-NLOG_DLL_SOURCE=NLog.dll
 LOW_PNG_SOURCE=icons/low.png
 
 endif
 
 if ENABLE_RELEASE
@@ -42,25 +40,19 @@
 RSS_PNG_SOURCE=icons/rss.png
 LOWEST_PNG_SOURCE=icons/lowest.png
 MONSOON_DESKTOP_SOURCE=monsoon.desktop
 DONOTDOWNLOAD_PNG_SOURCE=icons/donotdownload.png
 NORMAL_PNG_SOURCE=icons/normal.png
-NLOG_DLL_SOURCE=NLog.dll
 LOW_PNG_SOURCE=icons/low.png
 
 endif
 
 AL=al2
 SATELLITE_ASSEMBLY_NAME=.resources.dll
 
-UNSTABLE_LIBS = ./libs/MonoTorrent.dll \
-		./libs/Mono.Nat.dll
-
 PROGRAMFILES = \
-	$(MONSOON_EXE_CONFIG) \
-	$(NLOG_DLL) \
-	$(UNSTABLE_LIBS)
+	$(MONSOON_EXE_CONFIG)
 
 LINUX_DESKTOPAPPLICATIONS = \
 	$(MONSOON_DESKTOP)  
 
 PROGRAMFILES_ICONS = \
@@ -76,22 +68,11 @@
 BINARIES = \
 	$(MONSOON)  
 
 
 	
-all: update-libs $(ASSEMBLY) $(PROGRAMFILES) $(LINUX_DESKTOPAPPLICATIONS) $(PROGRAMFILES_ICONS) $(BINARIES) 
-	cp ./libs/MonoTorrent.dll ./bin/Release/MonoTorrent.dll
-	cp ./libs/Mono.Nat.dll ./bin/Release/Mono.Nat.dll
-
-./libs/Mono.Nat.dll: update-libs
-./libs/MonoTorrent.dll: update-libs
-
-update-libs:
-	mkdir -p ./libs
-	cp `pkg-config --variable=Libraries monotorrent` ./libs/MonoTorrent.dll
-	cp `pkg-config --variable=Libraries mono.nat` ./libs/Mono.Nat.dll
-
+all: $(ASSEMBLY) $(PROGRAMFILES) $(LINUX_DESKTOPAPPLICATIONS) $(PROGRAMFILES_ICONS) $(BINARIES) 
 
 FILES =  \
 	AboutDialog.cs \
 	AssemblyInfo.cs \
 	ByteConverter.cs \
@@ -184,16 +165,11 @@
 	Mono.Posix \
 	$(GCONF_SHARP_20_LIBS) \
 	$(GTK_SHARP_20_LIBS) \
 	$(GLIB_SHARP_20_LIBS)
 
-DLL_REFERENCES =  \
-		$(NLOG_DLL) \
-		./libs/Mono.Nat.dll \
-		./libs/MonoTorrent.dll
-
-CLEANFILES = $(PROGRAMFILES) $(LINUX_DESKTOPAPPLICATIONS) $(PROGRAMFILES_ICONS) $(BINARIES) $(UNSTABLE_LIBS)
+CLEANFILES = $(PROGRAMFILES) $(LINUX_DESKTOPAPPLICATIONS) $(PROGRAMFILES_ICONS) $(BINARIES)
 
 include $(top_srcdir)/Makefile.include
 
 MONSOON_EXE_CONFIG = $(BUILD_DIR)/Monsoon.exe.config
 HIGH_PNG = $(BUILD_DIR)/icons/high.png
@@ -203,11 +179,10 @@
 LOWEST_PNG = $(BUILD_DIR)/icons/lowest.png
 MONSOON_DESKTOP = $(BUILD_DIR)/monsoon.desktop
 DONOTDOWNLOAD_PNG = $(BUILD_DIR)/icons/donotdownload.png
 MONSOON = $(BUILD_DIR)/monsoon
 NORMAL_PNG = $(BUILD_DIR)/icons/normal.png
-NLOG_DLL = $(BUILD_DIR)/NLog.dll
 LOW_PNG = $(BUILD_DIR)/icons/low.png
 
 $(eval $(call emit-deploy-target,MONSOON_EXE_CONFIG))
 $(eval $(call emit-deploy-target,HIGH_PNG))
 $(eval $(call emit-deploy-target,HIGHEST_PNG))
@@ -216,18 +191,17 @@
 $(eval $(call emit-deploy-target,LOWEST_PNG))
 $(eval $(call emit-deploy-target,MONSOON_DESKTOP))
 $(eval $(call emit-deploy-target,DONOTDOWNLOAD_PNG))
 $(eval $(call emit-deploy-wrapper,MONSOON,monsoon,x))
 $(eval $(call emit-deploy-target,NORMAL_PNG))
-$(eval $(call emit-deploy-target,NLOG_DLL))
 $(eval $(call emit-deploy-target,LOW_PNG))
 
 
 $(build_xamlg_list): %.xaml.g.cs: %.xaml
 	xamlg '$<'
 
 $(build_resx_resources) : %.resources: %.resx
 	resgen2 '$<' '$@'
 
-$(ASSEMBLY) $(ASSEMBLY_MDB): $(build_sources) $(build_resources) $(build_datafiles) $(DLL_REFERENCES) $(PROJECT_REFERENCES) $(build_xamlg_list) $(build_satellite_assembly_list)
+$(ASSEMBLY) $(ASSEMBLY_MDB): $(build_sources) $(build_resources) $(build_datafiles) $(PROJECT_REFERENCES) $(build_xamlg_list) $(build_satellite_assembly_list)
 	mkdir -p $(dir $(ASSEMBLY))
-	$(ASSEMBLY_COMPILER_COMMAND) $(ASSEMBLY_COMPILER_FLAGS) -out:$(ASSEMBLY) -target:$(COMPILE_TARGET) $(build_sources_embed) $(build_resources_embed) $(build_references_ref)
+	$(ASSEMBLY_COMPILER_COMMAND) $(ASSEMBLY_COMPILER_FLAGS) -out:$(ASSEMBLY) -target:$(COMPILE_TARGET) $(build_sources_embed)  $(build_resources_embed) $(build_references_ref) -pkg:mono.nat -pkg:monotorrent -pkg:nlog
