Index: doc/Makefile.am
===================================================================
--- doc/Makefile.am	(revision 118618)
+++ doc/Makefile.am	(working copy)
@@ -32,7 +32,9 @@
 
 assemble: gtk-sharp-docs.zip gtk-sharp-docs.tree
 
-gtk-sharp-docs.zip gtk-sharp-docs.tree: $(srcdir)/en/*/*.xml $(srcdir)/en/*.xml
+gtk-sharp-docs.tree: gtk-sharp-docs.zip
+
+gtk-sharp-docs.zip: $(srcdir)/en/*/*.xml $(srcdir)/en/*.xml
 	$(MDASSEMBLER) --ecma $(srcdir)/en -o gtk-sharp-docs
 
 get-assemblies:
