--- a/mcs/class/System.Web/System.Web.Hosting/HostingEnvironment.cs
+++ b/mcs/class/System.Web/System.Web.Hosting/HostingEnvironment.cs
@@ -156,9 +156,18 @@
 			
 			HttpContext context = HttpContext.Current;
 			HttpRequest req = context == null ? null : context.Request;
-			if (req == null)
-				return null;
-
+			if (req == null) // ADDED for Orchard
+			{
+				var pp =  HostingEnvironment.ApplicationPhysicalPath;
+				String vp=virtualPath;
+				if (vp.IndexOf("~/") == 0)
+				{
+					vp = pp + virtualPath.Substring(2);
+				}
+				// TODO: if not "~/" ???? then what?
+				return vp;
+			} // END added for Orchard
+		        else
 			return req.MapPath (virtualPath);
 		}
 
