diff --git a/nuget/nunit.console.nuspec b/nuget/nunit.console.nuspec
index 7479f61..7c7e65f 100644
--- a/nuget/nunit.console.nuspec
+++ b/nuget/nunit.console.nuspec
@@ -25,8 +25,6 @@
     <file src="CHANGES.txt" />
     <file src="bin/nunit-agent.exe" target="tools" />
     <file src="bin/nunit-agent.exe.config" target="tools" />
-    <file src="bin/nunit-agent-x86.exe" target="tools" />
-    <file src="bin/nunit-agent-x86.exe.config" target="tools" />
     <file src="bin/nunit3-console.exe" target="tools" />
     <file src="bin/nunit3-console.exe.config" target="tools" />
     <file src="bin/nunit.engine.api.dll" target="tools" />
@@ -34,10 +32,6 @@
     <file src="bin/nunit.engine.dll" target="tools" />
     <file src="bin/nunit.engine.addins" target="tools" />
     <file src="bin/nunit.engine.addin.xml" target="tools" />
-    <file src="bin/Mono.Cecil.dll" target="tools" />
-    <file src="bin/addins/nunit.v2.driver.dll" target="tools/addins" />
-    <file src="bin/addins/nunit.core.dll" target="tools/addins" />
-    <file src="bin/addins/nunit.core.interfaces.dll" target="tools/addins" />
     <file src="bin/addins/nunit-v2-result-writer.dll" target="tools/addins" />
     <file src="bin/addins/nunit-project-loader.dll" target="tools/addins" />
     <file src="bin/addins/vs-project-loader.dll" target="tools/addins" />
