diff -u -r a/nuget/nunit.nuspec b/nuget/nunit.nuspec
--- a/nuget/nunit.nuspec	2015-12-02 05:52:00.000000000 +0300
+++ b/nuget/nunit.nuspec	2017-05-28 18:08:45.000000000 +0300
@@ -32,15 +32,7 @@
     <file src="LICENSE.txt" />
     <file src="NOTICES.txt" />
     <file src="CHANGES.txt" />
-    <file src="bin/net-2.0/nunit.framework.dll" target="lib/net20" />
-    <file src="bin/net-2.0/nunit.framework.xml" target="lib/net20" />
-    <file src="bin/net-4.0/nunit.framework.dll" target="lib/net40" />
-    <file src="bin/net-4.0/nunit.framework.xml" target="lib/net40" />
-    <file src="bin/net-4.5/nunit.framework.dll" target="lib/net45" />
-    <file src="bin/net-4.5/nunit.framework.xml" target="lib/net45" />
-    <file src="bin/portable/nunit.framework.dll" target="lib/dotnet" />
-    <file src="bin/portable/nunit.framework.xml" target="lib/dotnet" />
-    <file src="bin/portable/nunit.framework.dll" target="lib/portable-net45+win8+wp8+wpa81+Xamarin.Mac+MonoAndroid10+MonoTouch10+Xamarin.iOS10" />
-    <file src="bin/portable/nunit.framework.xml" target="lib/portable-net45+win8+wp8+wpa81+Xamarin.Mac+MonoAndroid10+MonoTouch10+Xamarin.iOS10" />
+    <file src="bin/$dir$/net-4.5/nunit.framework.dll" target="lib/net45" />
+    <file src="bin/$dir$/net-4.5/nunit.framework.xml" target="lib/net45" />
   </files>
 </package>
