diff -u -r a/nuget/nunit.engine.nuspec b/nuget/nunit.engine.nuspec
--- a/nuget/nunit.engine.nuspec	2015-12-02 05:52:00.000000000 +0300
+++ b/nuget/nunit.engine.nuspec	2017-05-28 19:01:55.000000000 +0300
@@ -21,12 +21,9 @@
     <file src="LICENSE.txt" />
     <file src="NOTICES.txt" />
     <file src="CHANGES.txt" />
-    <file src="bin\nunit.engine.dll" target="lib" />
-    <file src="bin\nunit.engine.api.dll" target="lib" />
-    <file src="bin\Mono.Cecil.dll" target="lib" />
-    <file src="bin\nunit-agent.exe" target="lib" />
-    <file src="bin\nunit-agent.exe.config" target="lib" />
-    <file src="bin\nunit-agent-x86.exe" target="lib" />
-    <file src="bin\nunit-agent-x86.exe.config" target="lib" />
+    <file src="bin/$dir$/nunit.engine.dll" target="lib" />
+    <file src="bin/$dir$/nunit.engine.api.dll" target="lib" />
+    <file src="bin/$dir$/nunit-agent.exe" target="lib" />
+    <file src="bin/$dir$/nunit-agent.exe.config" target="lib" />
   </files>
 </package>
\ No newline at end of file
