--- a/nuget/nunit.console.nuspec	2015-12-02 05:52:00.000000000 +0300
+++ b/nuget/nunit.console.nuspec	2017-05-28 18:49:12.746449899 +0300
@@ -23,23 +23,17 @@
     <file src="LICENSE.txt" />
     <file src="NOTICES.txt" />
     <file src="CHANGES.txt" />
-    <file src="bin/nunit-agent.exe" target="tools" />
-    <file src="bin/nunit-agent.exe.config" target="tools" />
-    <file src="bin/nunit-agent-x86.exe" target="tools" />
-    <file src="bin/nunit-agent-x86.exe.config" target="tools" />
-    <file src="bin/nunit3-console.exe" target="tools" />
-    <file src="bin/nunit3-console.exe.config" target="tools" />
-    <file src="bin/nunit.engine.api.dll" target="tools" />
-    <file src="bin/nunit.engine.api.xml" target="tools" />
-    <file src="bin/nunit.engine.dll" target="tools" />
-    <file src="bin/nunit.engine.addins" target="tools" />
-    <file src="bin/nunit.engine.addin.xml" target="tools" />
-    <file src="bin/Mono.Cecil.dll" target="tools" />
-    <file src="bin/addins/nunit.v2.driver.dll" target="tools/addins" />
-    <file src="bin/addins/nunit.core.dll" target="tools/addins" />
-    <file src="bin/addins/nunit.core.interfaces.dll" target="tools/addins" />
-    <file src="bin/addins/nunit-v2-result-writer.dll" target="tools/addins" />
-    <file src="bin/addins/nunit-project-loader.dll" target="tools/addins" />
-    <file src="bin/addins/vs-project-loader.dll" target="tools/addins" />
+    <file src="bin/$dir$/nunit-agent.exe" target="tools" />
+    <file src="bin/$dir$/nunit-agent.exe.config" target="tools" />
+    <file src="bin/$dir$/nunit3-console.exe" target="tools" />
+    <file src="bin/$dir$/nunit3-console.exe.config" target="tools" />
+    <file src="bin/$dir$/nunit.engine.api.dll" target="tools" />
+    <file src="bin/$dir$/nunit.engine.api.xml" target="tools" />
+    <file src="bin/$dir$/nunit.engine.dll" target="tools" />
+    <file src="bin/$dir$/nunit.engine.addins" target="tools" />
+    <file src="bin/$dir$/nunit.engine.addin.xml" target="tools" />
+    <file src="bin/$dir$/addins/nunit-v2-result-writer.dll" target="tools/addins" />
+    <file src="bin/$dir$/addins/nunit-project-loader.dll" target="tools/addins" />
+    <file src="bin/$dir$/addins/vs-project-loader.dll" target="tools/addins" />
   </files>
 </package>
