diff --git a/src/NUnitEngine/mono-addins/Mono.Addins.csproj b/src/NUnitEngine/mono-addins/Mono.Addins.csproj
index cb09546..7536d46 100644
--- a/src/NUnitEngine/mono-addins/Mono.Addins.csproj
+++ b/src/NUnitEngine/mono-addins/Mono.Addins.csproj
@@ -43,24 +41,6 @@
   </ItemGroup>
   <ItemGroup>
     <Compile Include="AssemblyInfo.cs" />
-    <Compile Include="Compatibility\Check.cs" />
-    <Compile Include="Compatibility\Enumerable.cs" />
-    <Compile Include="Compatibility\ExtensionAttribute.cs" />
-    <Compile Include="Compatibility\Funcs.cs" />
-    <Compile Include="Compatibility\Grouping.cs" />
-    <Compile Include="Compatibility\HashPrimeNumbers.cs" />
-    <Compile Include="Compatibility\HashSet.cs" />
-    <Compile Include="Compatibility\IGrouping.cs" />
-    <Compile Include="Compatibility\ILookup_T.cs" />
-    <Compile Include="Compatibility\IOrderedEnumerable_T.cs" />
-    <Compile Include="Compatibility\Lookup.cs" />
-    <Compile Include="Compatibility\OrderedEnumerable.cs" />
-    <Compile Include="Compatibility\OrderedSequence.cs" />
-    <Compile Include="Compatibility\QuickSort.cs" />
-    <Compile Include="Compatibility\SortContext.cs" />
-    <Compile Include="Compatibility\SortDirection.cs" />
-    <Compile Include="Compatibility\SortSequenceContext.cs" />
-    <Compile Include="Compatibility\Tuple.cs" />
     <Compile Include="Mono.Addins\ExtensionNode.cs" />
     <Compile Include="Mono.Addins\ExtensionTree.cs" />
     <Compile Include="Mono.Addins\TreeNode.cs" />
