diff --git a/src/Tasks/Microsoft.CSharp.targets b/src/Tasks/Microsoft.CSharp.targets
index 69994d0..0cc4320 100644
--- a/src/Tasks/Microsoft.CSharp.targets
+++ b/src/Tasks/Microsoft.CSharp.targets
@@ -16,6 +16,9 @@ Copyright (C) Microsoft Corporation. All rights reserved.
 -->
 
 <Project xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+   <PropertyGroup>
+      <CscToolPath>/usr/lib/mono/4.5</CscToolPath>
+   </PropertyGroup>
    <!-- 
         We are doing a cross-targeting build if there is no list of target frameworks specified
         nor is there a current target framework being built individually. In that case, this import is
